packages/stellar-axelar-std-derive/src/lib.rs

```diff
@@ -235,6 +236,65 @@ pub fn derive_its_executable(input: TokenStream) -> TokenStream {
     its_executable::its_executable(name).into()
 }
 
+#[proc_macro_attribute]
+pub fn only_owner(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as ItemFn);
+    let fn_vis = &input.vis;
+    let fn_sig = &input.sig;
+    let fn_block = &input.block;
+    let fn_inputs = &fn_sig.inputs;
+
+    // Check that env is the first parameter
+    let Some(syn::FnArg::Typed(pat_type)) = fn_inputs.first() else {
+        panic!("First parameter must be a typed parameter")
+    };
+    let syn::Pat::Ident(pat_ident) = &*pat_type.pat else {
+        panic!("First parameter must be a simple identifier")
+    };
+    assert!(
+        pat_ident.ident == "env",
+        "First parameter must be named 'env'"
+    );
+
+    let expanded = quote! {
+        #fn_vis #fn_sig {
+            Self::owner(&env).require_auth();
+            #fn_block
```

Comment:
```suggestion
            Self::owner(&env).require_auth();

            #fn_block
```
generated code is a bit more readable